docs/elasticsearch/logs.md

```diff
@@ -0,0 +1,173 @@
+[2019-10-18T16:52:51,395][DEBUG][o.e.i.r.TransportReindexAction] [node1] Freed [1] contexts
+```
+
+The DEBUG and TRACE levels are extremely verbose. If you enable either one to troubleshoot a problem, disable it after you finish.
+
+Other ways of changing log levels exist.
+
+1. Add lines to `elasticsearch.yml`:
+
+   ```yml
+   logger.org.elasticsearch.index.reindex: debug
+   ```
+
+   Modifying `elasticsearch.yml` makes the most sense if you want to reuse your logging configuration across multiple clusters or debug startup issues with a single node.
+
+2. Modify `log4j2.properties`:
+
+   ```
+   # Define a new logger with unique ID of reindex
+   logger.reindex.name = org.elasticsearch.index.reindex
+   # Set the log level for that ID
+   logger.reindex.level = debug
+   ```
+
+   This approach is extremely flexible, but requires familiarity with the [Log4j 2 property file syntax](https://logging.apache.org/log4j/2.x/manual/configuration.html#Properties). In general, the other options offer a simpler configuration experience.
+
+   If you examine the default `log4j2.properties` file in the configuration directory, you can see a few Elasticsearch-specific variables:
+
+   ```
+   appender.console.layout.pattern = [%d{ISO8601}][%-5p][%-25c{1.}] [%node_name]%marker %m%n
+   appender.rolling_old.fileName = ${sys:es.logs.base_path}${sys:file.separator}${sys:es.logs.cluster_name}.log
+   ```
+
+   - `${sys:es.logs.base_path}` is the directory for logs (e.g. `/var/log/elasticsearch/`).
+   - `${sys:es.logs.cluster_name}` is the name of the cluster.
+   - `[%node_name]` is the name of the node.
+
+
+## Slow logs
+
+Elasticsearch has two *slow logs*, logs that help you identify performance issues: the search slow log and the indexing slow log.
+
+These logs rely on thresholds to define what qualifies as a "slow" search or indexing operation. For example, you might decide that a query is slow if it takes more than 15 seconds to complete. By default, both logs are disabled (all thresholds set to `-1`):
+
+```json
+GET <some-index>/_settings?include_defaults=true
+
+{
+  "indexing": {
+    "slowlog": {
+      "reformat": "true",
+      "threshold": {
+        "index": {
+          "warn": "-1",
+          "trace": "-1",
+          "debug": "-1",
+          "info": "-1"
+        }
+      },
+      "source": "1000",
+      "level": "TRACE"
+    }
+  },
+  "search": {
+    "slowlog": {
+      "level": "TRACE",
+      "threshold": {
+        "fetch": {
+          "warn": "-1",
+          "trace": "-1",
+          "debug": "-1",
+          "info": "-1"
+        },
+        "query": {
+          "warn": "-1",
+          "trace": "-1",
+          "debug": "-1",
+          "info": "-1"
+        }
+      }
+    }
+  }
+}
+```
+
+To enable these logs, increase one or more thresholds:
+
+```json
+PUT <some-index>/_settings
+{
+  "indexing": {
+    "slowlog": {
+      "threshold": {
+        "index": {
+          "warn": "15s",
```

Comment:
Maybe this is obvious but you could explain this functionality to say that in this case if your index operation takes more than 750ms a trace log is generated and if it takes more than 3 seconds a debug log is generated and so on. Also, it would be useful to mention that If these are set to zero Elasticsearch logs all operations.